src/plugins/discover/public/application/main/components/layout/discover_main_content.tsx

```diff
@@ -145,7 +151,16 @@ export const DiscoverMainContent = ({
                 trackUiMetric={trackUiMetric}
               />
             </>
-          )}
+          ) : null}
+          {viewMode === VIEW_MODE.PATTERN_LEVEL ? (
```

Comment:
Why is the data reloaded and rerendered when the doc chart is shown / hidden?

Reply:
It causes an outer component rerender. If you're in the document or field stats tabs, they also reload.